python/lsst/atmospec/centroiding.py

```diff
@@ -0,0 +1,173 @@
+        storageClass="StructuredDataDict",
+        dimensions=("instrument", "visit", "detector"),
+    )
+
+
+class SingleStarCentroidTaskConfig(pipeBase.PipelineTaskConfig,
+                                   pipelineConnections=SingleStarCentroidTaskConnections):
+    """Configuration parameters for ProcessStarTask."""
+    astromRefObjLoader = pexConfig.ConfigurableField(
+        target=LoadIndexedReferenceObjectsTask,
+        doc="Reference object loader for astrometric calibration",
+    )
+    astrometry = pexConfig.ConfigurableField(
+        target=AstrometryTask,
+        doc="Task to perform astrometric calibration to refine the WCS",
+    )
+    qfmTask = pexConfig.ConfigurableField(
+        target=QuickFrameMeasurementTask,
+        doc="XXX",
+    )
+    referenceFilterOverride = pexConfig.Field(
+        dtype=str,
+        doc="Which filter in the reference catalog to match to?",
+        default="phot_g_mean"
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        # this is a null option now in Gen3 - do not set it here
+        # self.astromRefObjLoader.ref_dataset_name
+
+        self.astromRefObjLoader.pixelMargin = 1000
+
+        self.astrometry.wcsFitter.retarget(FitAffineWcsTask)
+        self.astrometry.referenceSelector.doMagLimit = True
+        magLimit = MagnitudeLimit()
+        magLimit.minimum = 1
+        magLimit.maximum = 15
+        self.astrometry.referenceSelector.magLimit = magLimit
+        self.astrometry.referenceSelector.magLimit.fluxField = "phot_g_mean_flux"
+        self.astrometry.matcher.maxRotationDeg = 5.99
+        self.astrometry.matcher.maxOffsetPix = 3000
+        self.astrometry.sourceSelector['matcher'].minSnr = 10
+
+
+class SingleStarCentroidTask(pipeBase.PipelineTask):
+    """XXX Docs here
+    """
+
+    ConfigClass = SingleStarCentroidTaskConfig
+    _DefaultName = 'singleStarCentroid'
+
+    def __init__(self, initInputs=None, **kwargs):
+        super().__init__(**kwargs)
+
+        self.makeSubtask("astrometry", refObjLoader=None)
+        self.makeSubtask('qfmTask')
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        refObjLoader = ReferenceObjectLoader(dataIds=[ref.datasetRef.dataId
+                                                      for ref in inputRefs.astromRefCat],
+                                             refCats=inputs.pop('astromRefCat'),
+                                             config=self.config.astromRefObjLoader, log=self.log)
+
+        refObjLoader.pixelMargin = 1000
+        self.astrometry.setRefObjLoader(refObjLoader)
+
+        # See L603 (def runQuantum(self, butlerQC, inputRefs, outputRefs):)
+        # in calibrate.py to put photocal back in
+
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, inputExp, inputSources):
+        """XXX Docs
+        """
+
+        # TODO: Change this to doing this the proper way
```

Comment:
Should this be ticketed so these TODO items can be tracked?  I'll not repeat this for other entries below, but the same question applies to them.

Reply:
Oh, they're ALL over this package, as it's not yet ready for prime-time. There's a catch-all ticket for that somewhere, but basically it's not efficient to ticket them all, especially as they'll all get done roughly at the same time.